### PR TITLE
feat:  check the namespace ID range of the leafHash in an absence proof to ensure soundness

### PR DESCRIPTION
**Pull request by staheri14**
## Overview

Addresses the issue set out in #115.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: spec/nmt.md [hunk 1]
```diff
@@ -126,9 +126,10 @@ As an example, the namespace proof for `nID = 0` for the NMT of Figure 1 (which
 
 ### Verification of NMT Absence Proof
 
-An NMT absence proof is deemed valid if 
-1) The verification of Merkle inclusion proof of the returned leaf is valid.  
-2) It satisfies the proof completeness as explained in the [Verification of NMT Inclusion Proof](#verification-of-nmt-inclusion-proof).
+An NMT absence proof is deemed valid if it meets the following:
+1) The minimum namespace ID of the leaf in the proof is greater than the queried `nID`.
```
Comment (rootulp) [inline spec/nmt.md:130]:
```suggestion
1) The namespace ID of the leaf in the proof is greater than the queried `nID`. 
```

Is it safe to remove "minimum" here? The leaf doesn't have a min/max namespace and instead it has a singular namespace ID

File: proof.go [hunk 2]
```diff
@@ -143,6 +143,13 @@ func (proof Proof) VerifyNamespace(h hash.Hash, nID namespace.ID, data [][]byte,
 	nIDLen := nID.Size()
 	if proof.IsOfAbsence() {
 		gotLeafHashes = append(gotLeafHashes, proof.leafHash)
+		// conduct some sanity checks:
+		leafMinNID := namespace.ID(proof.leafHash[:nIDLen])
+		if !nID.Less(leafMinNID) {
+			// leafHash.minNID  must be greater than nID
+			return false
+		}
```
Comment (rootulp) [inline proof.go:151]: [nit] is it safe to remove min in this context? AFAIK a leaf can only have a singular namespace ID

File: spec/nmt.md (hunk 1 above)
Comment (staheri14) [reply spec/nmt.md:130]: > The leaf doesn't have a min/max namespace and instead it has a singular namespace ID

Structurally, NMT leaves (but not their underlying data items) have two namespace IDs i.e., min and max, however, they carry identical values.

File: proof.go (hunk 2 above)
Comment (staheri14) [reply proof.go:151]: Can you please elaborate on your suggestion? Are you suggesting to change the naming of the variable `leafMinNID`?

Comment (rootulp) [reply proof.go:151]: Sorry I meant something like 
```suggestion
		// conduct some sanity checks:
		leafNID := namespace.ID(proof.leafHash[:nIDLen])
		if !nID.Less(leafNID) {
			// leafHash.NID  must be greater than nID
			return false
		}
```

but we can ignore given your other comment

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]
nice!

**Pull request merged**